sqlserver/datadog_checks/sqlserver/activity.py

```diff
@@ -53,16 +53,19 @@
     c.client_net_address as client_address,
     sess.host_name as host_name,
     r.*
-FROM sys.dm_tran_active_transactions at
-    INNER JOIN sys.dm_tran_session_transactions st ON st.transaction_id = at.transaction_id
-      LEFT OUTER JOIN sys.dm_exec_sessions sess ON st.session_id = sess.session_id
-    LEFT OUTER JOIN sys.dm_exec_connections c
+FROM sys.dm_exec_sessions sess
+    JOIN sys.dm_exec_connections c
         ON sess.session_id = c.session_id
-    LEFT OUTER JOIN sys.dm_exec_requests r
+    FULL OUTER JOIN sys.dm_exec_requests r
         ON c.connection_id = r.connection_id
+    FULL OUTER JOIN sys.dm_tran_session_transactions st ON st.session_id = sess.session_id
+    FULL OUTER JOIN sys.dm_tran_active_transactions at ON st.transaction_id = at.transaction_id
         CROSS APPLY sys.dm_exec_sql_text(c.most_recent_sql_handle) text
-    WHERE sess.session_id != @@spid
-    ORDER BY at.transaction_begin_time ASC
+WHERE sess.session_id != @@spid
+    AND (at.transaction_begin_time IS NOT NULL OR r.start_time IS NOT NULL)
```

Comment:
In order to capture both idle open transactions, and active sessions which are not transactions we need to do this check. If both are null that means the row is just an idle session that isnt doing anything